### ISSUE TITLE
Network settings should be loaded from EEPROM

### ISSUE THREAD

**Issue by JayFoxRox** [mborgerson/nxdk-rdt#10]
*This is more of an issue with nxdk itself, however, nxdk-rdt is directly affected by it.*

Currently, the network settings are hardcoded in net.c
For me, this meant that whenever I wanted to run nxdk-rdt in XQEMU, I had to go in and change the source code to use DHCP. When running on Xbox, I manually had to change it to use hardcoded addresses.

I've locally solved the issue by setting up a DHCP server for the ethernet port on my laptop, so the connected Xbox would also work with DHCP.
However, other people should not have to go through this.

We should fix this by loading the systems network settings from EEPROM.
The implementation should probably be in nxdk, but nxdk-rdt is currently the primary stakeholder. nxdk-rdt will also require some code changes. Hence the issue on this repository.

**Comment by mborgerson** [comment]
Good idea...configure your network settings for Xbox, let the titles use them as expected. I think the code really belongs in some kind of generic network startup function in nxdk. We can do it here for now since this is the only real application, but down the road it'll probably just get copy-pasted, so I'd rather it be in the common code.

**Comment by JayFoxRox** [comment]
This is also what I meant in my first post (see the first line and the last sentence). = This is an nxdk issue, but if the issue is more likely to be fixed if it's listed here. It should still be implemented directly in nxdk. However nxdk-rdt also needs a PR to support the changes in nxdk when this happens.

**Comment by JayFoxRox** [comment]
I've looked into this, and I can't seem to find any proper way to retrieve healthy, configurable information from EEPROM.
For reference, I've tried this:

```c
#define REG_DWORD 4
#define XC_ONLINE_IP_ADDRESS              0
#define XC_ONLINE_SUBNET_ADDRESS          1
#define XC_ONLINE_DEFAULT_GATEWAY_ADDRESS 2

    ULONG type;
    NTSTATUS hr;

    DWORD xbox_ipaddr;
    hr = ExQueryNonVolatileSetting(XC_ONLINE_IP_ADDRESS, &type, &xbox_ipaddr, sizeof(xbox_ipaddr), NULL);
    if (hr == STATUS_SUCCESS && type == REG_DWORD) { ip4_addr_set_u32(&ipaddr, xbox_ipaddr); }

    DWORD xbox_netmask;
    ExQueryNonVolatileSetting(XC_ONLINE_SUBNET_ADDRESS, &type, &xbox_netmask, sizeof(xbox_netmask), NULL);
    if (hr == STATUS_SUCCESS && type == REG_DWORD) { ip4_addr_set_u32(&netmask, xbox_netmask); }

    DWORD xbox_gw;
    ExQueryNonVolatileSetting(XC_ONLINE_DEFAULT_GATEWAY_ADDRESS, &type, &xbox_gw, sizeof(xbox_gw), NULL);
    if (hr == STATUS_SUCCESS && type == REG_DWORD) { ip4_addr_set_u32(&gw, xbox_gw); }
```

I'm not sure where Cxbx got their defines, but they don't seem to be working correctly for me (or do I do anything wrong?).

---

So EEPROM doesn't seem like a good choice. XBL seems to use the data stored in the XBL config sector.
That is the first couple of sectors on "\\Device\\Harddisk0\\partition0". In order for this to work, you have to have a XBL compatible Xbox though (= updated dash), otherwise you can't configure it.

I'm still not sure where homebrew dashboards store their config (evox, avalaunch, xbmc and unleashx are probably the most interesting targets).
